fe/fe-core/src/main/java/com/starrocks/scheduler/TaskManager.java

```diff
@@ -173,14 +238,213 @@ public ShowResultSet handleSubmitTaskStmt(SubmitTaskStmt submitTaskStmt) throws
             }
             throw new DdlException("Failed to create Task: " +  taskName + ", ErrorCode: " + createResult);
         }
-        String queryId = this.executeTask(taskName);
+        SubmitResult submitResult = executeTask(taskName);
+        if (submitResult.getStatus() != SubmitResult.SubmitStatus.Submitted) {
+            dropTasks(ImmutableList.of(task.getId()), false);
+        }
 
         ShowResultSetMetaData.Builder builder = ShowResultSetMetaData.builder();
         builder.addColumn(new Column("TaskName", ScalarType.createVarchar(40)));
         builder.addColumn(new Column("Status", ScalarType.createVarchar(10)));
-        List<String> item = ImmutableList.of(taskName, "Submitted");
+        List<String> item = ImmutableList.of(taskName, submitResult.getStatus().toString());
         List<List<String>> result = ImmutableList.of(item);
         return new ShowResultSet(builder.build(), result);
     }
 
+    public long loadTasks(DataInputStream dis, long checksum) throws IOException {
+        try {
+            readFields(dis);
+            LOG.info("finished replaying TaskManager from image");
+        } catch (EOFException e) {
+            LOG.info("no TaskManager to replay.");
+        }
+        return checksum;
+    }
+
+
+    public long saveTasks(DataOutputStream dos, long checksum) throws IOException {
+        write(dos);
+        return checksum;
```

Comment:
You should add some message to the checksum, such as task size. So that we can tell if the data is corrupted in advance.

Reply:
And update the checksum as the same logic when loading tasks